common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -567,6 +567,14 @@ def should_force_account_creation():
                 (current_provider.skip_email_verification or current_provider.send_to_registration_first))
 
     if not user:
+        details = kwargs.get('details', {})
+        if 'email' in details or 'username' in details:
+            # pylint: disable=invalid-name
+            qs = {'email': details['email']} if details.get('email') else \
+                {'username': details.get('username')}
+            if User.objects.filter(**qs).exists():
+                return dispatch_to_login()
```

Comment:
```
# Send the user to the login page if we already have an
# account that matches the authentication details.
user_queryset_filter = {}
email = details.get('email')
username= details.get('username')
if email:
    user_queryset_filter['email'] = email
elif username:
    user_queryset_filter['username'] = username

if user_queryset_filter and User.objects.filter(user_queryset_filter).exists():
    return dispatch_to_login()
```

Reply:
Create a `third_party_auth.utils` module and move the existing user check to a `def user_exists(details)` function there. Then you can reuse this code in the `login_and_registration_form` view.